### ISSUE TITLE
possible optimization?

### ISSUE THREAD

**Issue by zaggino** [aadsm/jschardet#14]
does it have to read the whole buffer?
https://github.com/aadsm/jschardet/blob/master/src/init.js#L72-L85

if it has to read the whole buffer, can it return a string as a result?

i want to check if file that i've opened is an `utf-8` file and then return it further, but I don't want to convert whole buffer to string twice

**Comment by aadsm** [comment]
Yes, you are right. There's no technical need to read the entire Buffer into a string. It's done this way because originally this library was only used on the web (before browsers implemented Blobs) and the conversion from Buffer to String was just a quick and dirty port to node.

**Comment by aadsm** [comment]
This is no longer an issue in `v4.0.0-rc.0`. `detect()`/`detectAll()` now also accepts `Uint8Array`.
